<?xml version="1.0"?>
<launch>
  <arg name="frequency"         default=""/>
  <arg name="topic_odometry"    default=""/>
  <arg name="topic_trajectory"  default=""/>
  <arg name="topic_mu"          default=""/>
  <arg name="Kp_ol"             default=""/>
  <arg name="Kd_ol"             default=""/>
  <arg name="Ki_ol"             default=""/>

  <node pkg="quad_control" type="OuterLoopControllerNode" name="OuterLoopControllerNode" output="screen">
    <param name="rate"  type="double"   value="$(arg frequency)"/>

    <param name="Kp"   type="double"   value="$(arg Kp_ol)"/>
    <param name="Ki"   type="double"   value="$(arg Kd_ol)"/>
    <param name="Kd"   type="double"   value="$(arg Ki_ol)"/>

    <remap from="/odometry"     to="$(arg topic_odometry)"/>
    <remap from="/trajectory"   to="$(arg topic_trajectory)"/>
    <remap from="/desiredMu"    to="$(arg topic_mu)"/>
  </node>

  <!--
  <node pkg="quad_control" type="Controller" name="Controller" output="screen">
    
    <param name="m"     type="double"   value="0.68"/>
    <param name="Ibx"   type="double"   value="0.007"/>
    <param name="Iby"   type="double"   value="0.007"/>
    <param name="Ibz"   type="double"   value="0.012"/>

    <param name="rate"  type="double"   value="$(arg frequency)"/>

    <param name="Kp"   type="double"   value="10"/>
    <param name="Kd"   type="double"   value="1" />
        
    <param name="K0"   type="double"   value="300"/>
    <param name="D0"   type="double"   value="1"/>
    <param name="v"     type="double"   value="100"/>
        
    <param name="ta1_wrench_estimator"  type="double"   value="0.01"/>
        
    <param name="kf1"    type="double"   value="100"/>
    <param name="kf2"    type="double"   value="100"/>

    <remap from="/odometry"     to="/$(arg mav_name)/odometryNED"/>
    <remap from="/trajectory"   to="/$(arg mav_name)/desiredTrajectory"/>
    <remap from="/commandData"  to="/$(arg mav_name)/command/motor_speed"/>
    <remap from="/logData"      to="/$(arg mav_name)/controller/logData"/>
  </node>
  -->
</launch>